py4cast/datasets/base.py

```diff
@@ -678,6 +684,131 @@ def summary(self):
                 print(table)  # Print the table
 
 
+@dataclass(slots=True)
+class Period:
+    start: dt.datetime
+    end: dt.datetime
+    step: int  # In hours, step btw the t0 of 2 samples
+    name: str
+
+    def __init__(self, start: int, end: int, step: int, name: str):
+        self.start = dt.datetime.strptime(str(start), "%Y%m%d%H")
+        self.end = dt.datetime.strptime(str(end), "%Y%m%d%H")
+        self.step = step
+        self.name = name
+
+    @property
+    def date_list(self):
```

Comment:
Does this list require to be an array of datetime.datetime? I mean do you have to use pandas just to do this?
If no, maybe you could use `numpy.timedelta64`.
Something like this is possible :
`return np.arange(self.start, self.end, np.timedelta64(self.step, "h"), dtype='datetime64[s]').tolist()`